src/avif.c

```diff
@@ -264,11 +272,27 @@ void avifImageSetMetadataXMP(avifImage * image, const uint8_t * xmp, size_t xmpS
     avifRWDataSet(&image->xmp, xmp, xmpSize);
 }
 
-void avifImageAllocatePlanes(avifImage * image, avifPlanesFlags planes)
+avifResult avifImageAllocatePlanes(avifImage * image, avifPlanesFlags planes)
 {
-    int channelSize = avifImageUsesU16(image) ? 2 : 1;
-    int fullRowBytes = channelSize * image->width;
-    int fullSize = fullRowBytes * image->height;
+    if (image->width == 0 || image->height == 0) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    if (image->depth != 8 && image->depth != 10 && image->depth != 12) {
+        return AVIF_RESULT_UNSUPPORTED_DEPTH;
+    }
+    const size_t channelSize = avifImageUsesU16(image) ? 2 : 1;
+    if (image->width > SIZE_MAX / channelSize) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    const size_t fullRowBytes = channelSize * image->width;
+    if ((image->height > SIZE_MAX / fullRowBytes) || (fullRowBytes > UINT_MAX)) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    const size_t fullSize = fullRowBytes * image->height;
+    if (image->width * image->height > AVIF_DEFAULT_IMAGE_SIZE_LIMIT) {
```

Comment:
We should not do this check here. One reason is that we have performed a stronger check above. Another reason is that the image size limit is a configurable decoder option, and `AVIF_DEFAULT_IMAGE_SIZE_LIMIT` is merely the default value of that option.

Reply:
Done. I had to add an extra conversion to `uint64_t` for `shiftedW/H`. Also [`decoder->imageSizeLimit` cannot be greater than `AVIF_DEFAULT_IMAGE_SIZE_LIMIT` during decoding for now](https://github.com/AOMediaCodec/libavif/blob/1d2462a20d7140ccfb4edd659ad38eea05aab41a/src/read.c#L3132).